src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -74,319 +76,294 @@ const defaultProps = {
+                                    return;
+                                }
+
+                                if (messageEditInput === relatedTargetId) {
+                                    return;
+                                }
+                                openReportActionComposeViewWhenClosingMessageEdit();
+                            }}
+                            selection={selection}
+                            onSelectionChange={(e) => setSelection(e.nativeEvent.selection)}
+                        />
+                    </View>
+                    <View style={styles.editChatItemEmojiWrapper}>
+                        <EmojiPickerButton
+                            isDisabled={props.shouldDisableEmojiPicker}
+                            onModalHide={() => InteractionManager.runAfterInteractions(() => textInputRef.current.focus())}
+                            onEmojiSelected={addEmojiToTextBox}
+                            nativeID={emojiButtonID}
+                        />
+                    </View>
+
+                    <View style={styles.alignSelfEnd}>
+                        <Tooltip text={props.translate('common.saveChanges')}>
+                            <PressableWithFeedback
+                                style={[styles.chatItemSubmitButton, hasExceededMaxCommentLength ? {} : styles.buttonSuccess]}
+                                onPress={publishDraft}
+                                nativeID={saveButtonID}
+                                disabled={hasExceededMaxCommentLength}
+                                accessibilityRole="button"
+                                accessibilityLabel={props.translate('common.saveChanges')}
+                                hoverDimmingValue={1}
+                                pressDimmingValue={0.2}
+                            >
+                                <Icon
+                                    src={Expensicons.Checkmark}
+                                    fill={hasExceededMaxCommentLength ? themeColors.icon : themeColors.textLight}
+                                />
+                            </PressableWithFeedback>
+                        </Tooltip>
                     </View>
                 </View>
-                <ExceededCommentLength
-                    comment={this.state.draft}
-                    onExceededMaxCommentLength={this.setExceededMaxCommentLength}
-                />
-            </>
-        );
-    }
+            </View>
+            <ExceededCommentLength
+                comment={draft}
+                onExceededMaxCommentLength={(hasExceeded) => setHasExceededMaxCommentLength(hasExceeded)}
+            />
+        </>
+    );
 }
 
 ReportActionItemMessageEdit.propTypes = propTypes;
 ReportActionItemMessageEdit.defaultProps = defaultProps;
+ReportActionItemMessageEdit.displayName = 'ReportActionItemMessageEdit';
 export default compose(
     withLocalize,
     withWindowDimensions,
```

Comment:
Let's also use `useWindowDimensions` hook from `src/hooks` instead of `withWindowDimensions`. Also don't forget to remove the unused props from these HOCs when you remove them

Reply:
Lastly, let's also add a very simple `useKeyboardState` hook in `src/hooks`:

```js
export default function useKeyboardState() {
    const [isKeyboardShown, setIsKeyboardShown] = useState(false);
    useEffect(() => {
        const keyboardDidShowListener = Keyboard.addListener('keyboardDidShow', () => setIsKeyboardShown(true));
        return keyboardDidShowListener.remove;
    }, []);
    useEffect(() => {
        const keyboardDidHideListener = Keyboard.addListener('keyboardDidHide', () => setIsKeyboardShown(false));
        return keyboardDidHideListener.remove;
    }, []);
    return {
        isKeyboardShown,
    };
}
```

Then use it like this:

```js
const {isKeyboardShown} = useKeyboardState();
```

And then remove the `withKeyboardState` HOC and props.

Reply:
We should get used to using hooks more as we migrate to a functional react style because they're often simpler to read, simpler to use, simpler to type once we start migrating everything to TypeScript, and don't bloat the component tree as much as HOCs.

Reply:
Actually, I see we have a context for `withKeyboardState` that's preventing us from setting up duplicate listeners. So the `useKeyboardState` hook becomes even simpler. Just:

1. Export `KeyboardStateContext` from `withKeyboardState`
1. Implement `useKeyboardState` like so:

    ```js
    import {useContext} from 'react';
    import {KeyboardStateContext} from '../components/withKeyboardState';
    
    export default function useKeyboardState() {
        return useContext(KeyboardStateContext);
    }
    ```